assets/js/atomic/blocks/product-elements/image/style.scss

```diff
@@ -57,3 +55,13 @@
 .is-loading .wc-block-components-product-image {
 	@include placeholder();
 }
+
+.wc-block-components-product-image {
+	margin: 0 0 $gap-small;
+}
+
+
+// Don't use global style while skeleton is visible.
```

Comment:
I was wondering if we should honor global styles also while the block is loading. Otherwise there is a visual jump:

https://user-images.githubusercontent.com/3616980/155564392-32ca4096-50c0-4768-875f-df21616a00b6.mp4

Reply:
If I comment this code when I edit the margin with the Global Style, we have this UI (I think that it is worse):

![image](https://user-images.githubusercontent.com/4463174/155571358-dec08ac0-c6fe-42da-a3f7-8c6cd7b7027a.png)
![image](https://user-images.githubusercontent.com/4463174/155571593-467ffbdd-9ff8-4a97-8521-41cee8a66b17.png)

GIF
![Screen Capture on 2022-02-24 at 18-01-53](https://user-images.githubusercontent.com/4463174/155571854-a4068250-337a-41b5-be33-1cf8af0b715e.gif)

Reply:
Oh, interesting. I think the issue here comes from the `placeholder()` styles, which have `width: 100%`:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/f4a11f158c5efd45f2caeeb9e67a0bbc61de0f94/assets/js/atomic/blocks/product-elements/image/style.scss#L58

Could we just add a `width: auto;` declaration after it? Besides that, you would also need to pass `spacingProps.style` to the [placeholder div](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/f4a11f158c5efd45f2caeeb9e67a0bbc61de0f94/assets/js/atomic/blocks/product-elements/image/block.js#L56-L65). It seemed to work fine for me:

https://user-images.githubusercontent.com/3616980/155696490-686165e2-8754-4522-92c0-1a18a5dfca4f.mp4

You could also pass `borderProps.style` to get the border radius to be applied as well.

Reply:
Oh! Thanks for the hint! 

I update the mixin too because the borderProps works only for block-level style.
With the global style, the margin value of the mixin has more priority than the global style value.
I added an optional parameter that allows to not include the border-radius value.


![image](https://user-images.githubusercontent.com/4463174/155733103-cacc467f-5172-421b-898c-37223d039300.png)
*before updating mixin function*

Reply:
> With the global style, the margin value of the mixin has more priority than the global style value.

Good catch!